cmd/api-response.go

```diff
@@ -702,6 +702,18 @@ func writeErrorResponseJSON(ctx context.Context, w http.ResponseWriter, err APIE
 	writeResponse(w, err.HTTPStatusCode, encodedErrorResponse, mimeJSON)
 }
 
+// writeVersionMismatchResponse - writes custom error responses for version mismatches.
+func writeVersionMismatchResponse(ctx context.Context, w http.ResponseWriter, err APIError, reqURL *url.URL, isJSON bool) {
+	// Generate error response.
+	errorResponse := getAPIErrorResponse(ctx, err, reqURL.String(), w.Header().Get(xhttp.AmzRequestID), globalDeploymentID)
+	encodedErrorResponse := encodeResponseJSON(errorResponse)
+	if isJSON {
+		writeResponse(w, err.HTTPStatusCode, encodedErrorResponse, mimeJSON)
+	} else {
+		writeResponse(w, err.HTTPStatusCode, encodedErrorResponse, mimeNone)
```

Comment:
you shouldn't set mimeNone for JSON body. It should be 

```suggestion
		writeResponse(w, err.HTTPStatusCode, []byte(errorResponse.Description), mimeNone)
```